Amazon.IonHashDotnet/Serializer.cs

```diff
@@ -69,16 +69,17 @@ internal static byte[] Escape(byte[] bytes)
             return bytes;
         }
 
-        internal virtual void Scalar(IIonHashValue ionValue)
+        internal virtual void Scalar(IIonHashValue ionValue, bool isInStruct)
         {
             this.HandleAnnotationsBegin(ionValue);
             this.BeginMarker();
 
             dynamic ionValueValue = ionValue.CurrentIsNull ? null : ionValue.CurrentValue;
+            IonType ionType = ionValue.CurrentIsNull ? IonType.None : ionValue.CurrentType;
```

Comment:
Does this work as it should?

For example, using a ValueFactory and calling NewNullInt(), then passing it here, what happens, and what should happen?

Reply:
So for null int, GetBytes() below returns byte array of size 1.

ScalarOrNullSplitParts() has the following conditional `if (type == IonType.Int && bytes.Length > offset)`. Currently bytes.Length > offset is false so it will never get into the conditional even if type is IonType.Int.

I will revert changing the ionType and pass ionValue.CurrentIsNull into ScalarOrNullSplitParts() to make it less confusing